### PR TITLE
ENH add ncores field to asv-machine

### PR DESCRIPTION
**Pull request by pierreglaser**
Simple PR to add the number of CPUs into `.asv-machine.json`

### CODE REVIEW

File: asv/machine.py [hunk 1]
```diff
@@ -144,6 +149,7 @@ def get_defaults():
         return {
             'machine': node,
             'os': "{0} {1}".format(system, release),
+            'ncores': os.cpu_count(),
```
Comment (pv) [inline asv/machine.py]: Should work on Python 2 still.
Also, should be `six.text_type`, not int.

File: asv/machine.py [hunk 2]
```diff
@@ -119,6 +119,11 @@ class Machine(object):
          including its speed and class.  For example, 'Intel(R)
          Core(TM) i5-2520M CPU @ 2.50GHz (4 cores)'."""),
 
+        ("ncores",
```
Comment (pv) [inline asv/machine.py]: `num_cpu`?

Comment (pierreglaser) [reply asv/machine.py]: Actually I like more num_cpu, but it felt weird to use underscores in command line options.

**Comment by pv** [comment]
thanks, merged (build failures due to conda flaking out again)

**Pull request merged**